src/quoter.ts

```diff
@@ -42,33 +48,37 @@ export abstract class SwapQuoter {
     const singleHop = route.pools.length === 1
     const quoteAmount: string = toHex(amount.quotient)
     let calldata: string
+    const swapInterface = options.useQuoterV2 ? new Interface(IQuoterV2.abi) : this.INTERFACE
 
     if (singleHop) {
-      if (tradeType === TradeType.EXACT_INPUT) {
-        calldata = SwapQuoter.INTERFACE.encodeFunctionData(`quoteExactInputSingle`, [
-          route.tokenPath[0].address,
-          route.tokenPath[1].address,
-          route.pools[0].fee,
-          quoteAmount,
-          toHex(options?.sqrtPriceLimitX96 ?? 0)
-        ])
-      } else {
-        calldata = SwapQuoter.INTERFACE.encodeFunctionData(`quoteExactOutputSingle`, [
-          route.tokenPath[0].address,
-          route.tokenPath[1].address,
-          route.pools[0].fee,
-          quoteAmount,
-          toHex(options?.sqrtPriceLimitX96 ?? 0)
-        ])
+      const quoteV2Params = {
+        tokenIn: route.tokenPath[0].address,
+        tokenOut: route.tokenPath[1].address,
+        // property name 'amount' is suffixed with 'in' for quoteExactInputSingle
+        ...(options.useQuoterV2
+          ? tradeType === TradeType.EXACT_INPUT
+            ? { amountIn: quoteAmount }
+            : { amount: quoteAmount }
+          : { fee: route.pools[0].fee }),
+        ...(options.useQuoterV2 ? { fee: route.pools[0].fee } : { amount: quoteAmount }),
+        sqrtPriceLimitX96: toHex(options?.sqrtPriceLimitX96 ?? 0)
       }
+      const tradeTypeFunctionName =
+        tradeType === TradeType.EXACT_INPUT ? 'quoteExactInputSingle' : 'quoteExactOutputSingle'
```

Comment:
Would you mind doing the same thing for the call below?
(`quoteExactOutput` and `quoteExactInput`)